rtl8720dn/adapter.go

```diff
@@ -1,25 +1,52 @@
 package rtl8720dn
 
 import (
+	"machine"
 	"time"
 
 	"tinygo.org/x/drivers/net"
 )
 
-func (r *RTL8720DN) ConnectToAccessPoint(ssid, pass string, timeout time.Duration) error {
+type Driver struct {
+	*RTL8720DN
+}
+
+func New(uart *machine.UART) *Driver {
+	return &Driver{
+		RTL8720DN: &RTL8720DN {
+			port:  uart,
+			seq:   1,
+			sema:  make(chan bool, 1),
+			debug: false,
+		},
+	}
+}
```

Comment:
The CHIP_PU pin is required to initialize the rtl8720dn even if it is not a Wio Terminal.
So I think the following is correct for the New function.

```
func New(uart *machine.UART, en machine.Port) *Driver
```

![image](https://user-images.githubusercontent.com/9251039/187435082-b592b87f-b388-4039-8eda-ce87f22630b8.png)

```

Reply:
Thanks for the excellent clarification @sago35 

Please see my latest commit that I think implements your feedback

Reply:
@sago35 if you think my changes are correct now, I will proceed to complete modifying the other rtl8720dn examples in the same way.

Reply:
@deadprogram 
I will investigate the source code some more.

It looks like it needs to call UART.Configure after manipulating the CHIP_PU.
So I think we have to pass machine.Pin before uart.Configure to initialize it.

Reply:
@deadprogram 
I thought about it again.
Two major tasks were undertaken.
If it looks OK, `git cherry-pick` what you need.

1. remove code for wioterminal from driver
2. [UART3 defined in machine/board_wioterminal.go](https://github.com/tinygo-org/tinygo/pull/3110)

https://github.com/tinygo-org/drivers/tree/rtl8720dn-device-wrapper-sago35

Reply:
I made some changes inspired by, but not exactly the same, as what you did. I still really wanted to keep the default UART config for Wioterminal in the driver to reduce the amount of code needed for applications.

See what you think @sago35

Reply:
Any other feedback @sago35 or can I now proceed with the rest of the examples?